tests/integ/modin/index/test_timedelta_index_methods.py

```diff
@@ -0,0 +1,59 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import modin.pandas as pd
+import pandas as native_pd
+import pytest
+
+import snowflake.snowpark.modin.plugin  # noqa: F401
+from tests.integ.modin.sql_counter import sql_count_checker
+
+
+@sql_count_checker(query_count=3)
+def test_timedelta_index_construction():
+    # create from native pandas timedelta index.
+    index = native_pd.TimedeltaIndex(["1 days", "2 days", "3 days"])
+    snow_index = pd.Index(index)
+    assert isinstance(snow_index, pd.TimedeltaIndex)
+
+    # create from query compiler with timedelta type.
+    df = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]}, index=index)
+    snow_index = df.index
+    assert isinstance(snow_index, pd.TimedeltaIndex)
+
+    # create from snowpark pandas timedelta index.
+    snow_index = pd.Index(pd.TimedeltaIndex([123]))
+    assert isinstance(snow_index, pd.TimedeltaIndex)
```

Comment:
Now that `set_index_from_columns` should handle Timedelta, can we add a test for something like this:

```
date_range1 = pd.date_range("2000-01-01", periods=10, freq="h")
date_range2 = pd.date_range("2001-05-01", periods=10, freq="h")
date_range2 - date_range1
```

Reply:
Ack